emsdk.py

```diff
@@ -784,7 +784,7 @@ def GIT(must_succeed=True):
       if ret == 0:
         cached_git_executable = git
         return git
-    except:
+    except Exception:
```

Comment:
When we are trying to catch everything, why not use the shorter form?

Reply:
% [`ruff rule E722`](https://docs.astral.sh/ruff/rules/bare-except)
## Why is this bad?
A bare `except` catches `BaseException` which includes `KeyboardInterrupt`, `SystemExit`, `Exception`, and others. Catching `BaseException` can make it hard to interrupt the program (e.g., with Ctrl-C) and can disguise other problems.

---

https://peps.python.org/pep-0008/#programming-recommendations
> When catching exceptions, mention specific exceptions whenever possible instead of using a bare except: clause:
```python
try:
    import platform_specific_module
except ImportError:
    platform_specific_module = None
```
> A bare except: clause will catch SystemExit and KeyboardInterrupt exceptions, making it harder to interrupt a program with Control-C, and can disguise other problems. If you want to catch all exceptions that signal program errors, use except Exception: (bare except is equivalent to except BaseException:).

> A good rule of thumb is to limit use of bare ‘except’ clauses to two cases:
> * If the exception handler will be printing out or logging the traceback; at least the user will be aware that an error has occurred.
> * If the code needs to do some cleanup work, but then lets the exception propagate upwards with raise. try...finally can be a better way to handle this case.

Reply:
Ah yes we don't want to catch KeyboardInterrupt here I guess